munit/native/src/main/scala/munit/internal/PlatformCompat.scala

```diff
@@ -7,10 +7,17 @@ import scala.scalanative.reflect.Reflect
 import sbt.testing.Task
 import sbt.testing.EventHandler
 import sbt.testing.Logger
+import scala.concurrent.Await
+import scala.concurrent.Awaitable
 import scala.concurrent.duration.Duration
 import scala.concurrent.ExecutionContext
 
 object PlatformCompat {
+  def awaitResult[T](awaitable: Awaitable[T]): T = {
+    scalanative.runtime.loop()
+    Await.result(awaitable, Duration.Inf)
+  }
```

Comment:
This is how specs2 is working as well.
https://github.com/etorreborre/specs2/blob/b4fb0a16f1a0369fe69eb994220317a5aaf2f3d9/common/native/src/main/scala/org/specs2/concurrent/package.scala#L8-L11